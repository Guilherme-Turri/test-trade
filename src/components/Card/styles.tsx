import styled, { css } from 'styled-components'
import { Container as TC } from '../TextComponent/styles'

interface availableProps {
  available: boolean
}

export const Container = styled.div<availableProps>`
${({ theme, available }) => css`
background: white;
display: flex;
flex-direction: column;
justify-content: space-evenly;
align-items: center;
height: 15.625rem;
width: 15.625rem;
border-radius: 5px;
margin: ${theme.spacings.normal};
box-shadow: ${available ? '0 0 5px ' + theme.color.secondaryColor : 'none'};
cursor: ${available ? 'pointer' : ''};

${TC}{
  color: ${available ? '' : 'grey'};
  font-family: ${theme.fonts.logoFont};
  border-bottom: ${available ? '1px solid ' + theme.color.secondaryColor : 'none'};
  padding: ${theme.spacings.xsmall};;
}
`}
`
export const Img = styled.div`
${({ theme }) => css`
height: 6.25rem;
width: 6.25rem;
`}
`