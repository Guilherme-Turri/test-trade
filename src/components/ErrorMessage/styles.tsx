import styled, { css } from 'styled-components'

export const Container = styled.p`
${({ theme }) => css`
color: ${theme.color.error};
font-size: 12px;
`}
`