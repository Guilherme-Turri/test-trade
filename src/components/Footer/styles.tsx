import styled, { css } from 'styled-components'
import wave from '../../UI/assets/wave.svg'

export const Footer = styled.footer`
${({ theme }) => css`
  color:${theme.color.backgroundColor};
  font-weight: bold;
  padding: 5px;
  display: flex;
  justify-content: center;
  align-items: end;
  display: flex;
  position: fixed;
  bottom: 0; 
  background-image: url(${wave});
  background-repeat: no-repeat;
  background-size: cover;
  width:100%;
  height: 320px;
  z-index: -10000;
`}
`