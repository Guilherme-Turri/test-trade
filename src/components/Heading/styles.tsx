import styled, { css } from 'styled-components'
import { HeadingProps } from './index'

export const Container = styled.h1<HeadingProps>`
  ${({ theme, uppercase, big }) => css`
      text-transform: ${uppercase ? 'uppercase' : 'none'};
      color: ${theme.color.secondaryColor};
      font-size: ${big ? '60px' : '20px'};
      font-family: ${theme.fonts.logoFont};
      `}
  `