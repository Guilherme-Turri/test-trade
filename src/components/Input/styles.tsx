import styled, { css } from "styled-components";

export const Container = styled.input`
${({ theme }) => css`
outline: none;
all: unset;
border: 1px solid ${theme.color.secondaryColor};
border-radius: ${theme.radius.small};
width: 100%;
height: 1.563rem;
padding: ${theme.spacings.normal};
box-sizing: border-box;
color: ${theme.color.primaryColor};
::placeholder{
  color:'#95A5A0'!important;
}
`}
`