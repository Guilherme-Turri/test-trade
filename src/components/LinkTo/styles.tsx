import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    color: ${theme.color.secondaryColor};
  cursor: pointer;
  text-decoration: none;
  `}
`;