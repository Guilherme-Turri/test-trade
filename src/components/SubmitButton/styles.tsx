import styled, { css } from 'styled-components'
interface ContainerProps {
  enableColor: boolean
}
export const Container = styled.button<ContainerProps>`
${({ theme, enableColor }) => css`
  display: flex;
  all:unset;
  background: ${enableColor ? theme.color.secondaryColor : '#CACCb1'};
  width: 100%;
  border-radius:  5px;
  height: 2rem;
  text-align: center;
  color:${theme.color.backgroundColor};
  `}
`