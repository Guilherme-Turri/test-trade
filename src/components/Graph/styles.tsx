import styled, { css } from 'styled-components'
import { Container as HeadingContainer } from '../Heading/styles'

export const Container = styled.section`
${({ theme }) => css`
display: flex;
flex-direction:column;
justify-content:center;
align-items:center;
width:100%;
height:100%;
z-index:-999999999;
${HeadingContainer}{
  margin-bottom: ${theme.spacings.normal};
}
`}
`