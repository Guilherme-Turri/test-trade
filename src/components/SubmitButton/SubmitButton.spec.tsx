import { componentRender } from "../../render/componentRender";
import { screen } from '@testing-library/react'
import { SubmitButton } from ".";
import { theme } from "../../styles/theme";

describe('<SubmitButton/>', () => {

  it('should render on screen properly', () => {
    const { container } = componentRender(<SubmitButton>btn</SubmitButton>)
    const btn = screen.getByText('btn')
    expect(btn).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
  it('should render on screen enable', () => {
    componentRender(<SubmitButton apiKey="123456" userName="test" error="">btn</SubmitButton>)
    const btn = screen.getByText('btn')
    expect(btn).toHaveStyleRule('color', theme.color.backgroundColor);
  })
})