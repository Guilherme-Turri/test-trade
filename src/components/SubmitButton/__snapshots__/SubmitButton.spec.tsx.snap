// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton/> should render on screen properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  all: unset;
  background: #CACCb1;
  width: 100%;
  border-radius: 5px;
  height: 2rem;
  text-align: center;
  color: #F3F4F5;
  cursor: pointer;
}

<div>
  <button
    class="c0"
  >
    btn
  </button>
</div>
`;
